luigi_pipeline/sex_related_checks.py

```diff
@@ -0,0 +1,568 @@
+            plt.savefig(out)
+
+        return sex_ht
+
+    def get_chr_cov(
+        self,
+        mt: hl.MatrixTable,
+        chr_name: str,
+        call_rate_threshold: float = 0.25,
+        af_threshold: float = 0.01,
+        af_field: str = "AF",
+    ) -> hl.Table:
+        """
+        Calculate mean chromosome coverage.
+        :param mt: MatrixTable containing samples with chrY variants
+        :param build: Reference used, either GRCh37 or GRCh38
+        :param chr_name: Chosen chromosome. Must be either autosome (number only) or sex chromosome (X, Y)
+        :param call_rate_threshold: Minimum call rate threshold. Default is 0.25
+        :param af_threshold: Minimum allele frequency threshold. Default is 0.01
+        :param af_field: Name of field containing allele frequency information. Default is "AF"
+        :return: Table annotated with mean coverage of specified chromosome
+        """
+        logger.info(
+            "Filtering to chromosome (and filtering to non-par regions if chromosome is X or Y)..."
+        )
+        ref_genome = mt.locus.dtype.reference_genome.name
+
+        if chr_name == "Y":
+            chr_place = 23
+        elif chr_name == "X":
+            chr_place = 22
+        else:
+            try:
+                # Chromosome index in '.contigs' list should be one less than the chromosome number
+                chr_place = int(chr_name) - 1
+            except ValueError:
+                logger.error("chr_name cannot be converted to an integer")
+                return -99
+
+        chr_name = hl.get_reference(ref_genome).contigs[chr_place]
+        sex_mt = hl.filter_intervals(
+            mt,
+            [hl.parse_locus_interval(chr_name, reference_genome=ref_genome)],
+        )
+
+        if chr_place == 22:
+            sex_mt = sex_mt.filter_rows(sex_mt.locus.in_x_nonpar())
+        if chr_place == 23:
+            sex_mt = sex_mt.filter_rows(sex_mt.locus.in_y_nonpar())
+
+        # Filter to common SNVs above defined callrate (should only have one index in the array because the MT only contains biallelic variants)
+        # TODO: Make callrate filtering optional before adding code to gnomad_methods
+        sex_mt = sex_mt.filter_rows(sex_mt[af_field] > af_threshold)
+        sex_mt = hl.variant_qc(sex_mt)
+        sex_mt = sex_mt.filter_rows(sex_mt.variant_qc.call_rate > call_rate_threshold)
+
+        logger.info("Returning mean coverage on chromosome %s...", chr_name)
+        sex_mt = sex_mt.annotate_cols(**{f"{chr_name}_mean_dp": hl.agg.mean(sex_mt.DP)})
+        return sex_mt.cols()
+
+    def annotate_globals(self, mt):
+        return mt.annotate_globals(
+            sourceFilePath=",".join(self.source_paths),
+            genomeVersion=self.genome_version,
+            sampleType=self.sample_type,
+            hail_version=pkg_resources.get_distribution("hail").version,
+        )
+
+    def read_input_write_mt(self):
```

Comment:
Should this function be calling `call_sex()`?

In our other tasks `read_input_write_mt` does the bulk of the work and then writes the final output file.